docs/src/test/count_outfiles.py

```diff
@@ -0,0 +1,31 @@
+import os
+import sys
+
+# Test objective: To count the number of expected files in man2/man3
+
+# This script mimics the build process and checks expected output.
+path = os.path.realpath("md_roff_compat.py")
+docs_dir = os.path.dirname(os.path.dirname(os.path.dirname(path)))
+os.chdir(docs_dir)
+os.system("make clean -s && make preprocess -s")
+os.system("make all -j16 -s")
```

Comment:
probably don't want to hardcode the -j16 into the test

Reply:
I understand, the reason I added that in was to speed up docs compilation which can be highly parallel. Will change to something similar to what we do in `./etc/Build.sh`